nsv13/code/game/gamemodes/overmap/objectives/system_defence_armada.dm

```diff
@@ -20,6 +20,12 @@
 		return
 
 	var/datum/star_system/target = SSstar_system.system_by_id(selected_system)
-	if(target.alignment != "nanotrasen")
+	var/obj/structure/overmap/OM = SSstar_system.find_main_overmap()
+	if(target.alignment != "nanotrasen" && OM.current_system != target)
 		priority_announce("Attention [station_name()]. Our presence in this sector has been severely diminished due to your incompetence. Return to base immediately for disciplinary action.", "Naval Command")
+		status = 2
 		SSovermap_mode.mode.defeat()
+
+	if(target.enemies_in_system.len == 0)
```

Comment:
Generally speaking, you should use `length()` when reading length and `.len` when modifying it. length() is faster
```suggestion
	if(!length(target.enemies_in_system))
```